upload/upload_test.go

```diff
@@ -0,0 +1,131 @@
+package upload
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type TestSuite struct {
+	suite.Suite
+}
+
+func TestConfigTestSuite(t *testing.T) {
+	suite.Run(t, new(TestSuite))
+}
+
+func (suite *TestSuite) SetupTest() {
+
+}
+
+func (suite *TestSuite) TestConfigNoFile() {
+
+	configPath := "nofile.conf"
+
+	_, err := loadConfigFile(configPath)
+	assert.EqualError(suite.T(), err, "open nofile.conf: no such file or directory")
+}
+
+func (suite *TestSuite) TestConfigWrongFile() {
+	var confFile = `
+access_token = someToken
+access_key = someUser
+host_bucket = someHostBase
+guess_mime_type!True
+encrypt = False
+`
+
+	configPath, err := ioutil.TempFile(os.TempDir(), "s3cmd-")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer os.Remove(configPath.Name())
+
+	err = ioutil.WriteFile(configPath.Name(), []byte(confFile), 0644)
+	if err != nil {
+		log.Fatalf("failed to write temp config file, %v", err)
+	}
+
+	_, err = loadConfigFile(configPath.Name())
+	assert.EqualError(suite.T(), err, "key-value delimiter not found: guess_mime_type!True\n")
+}
+
+func (suite *TestSuite) TestConfigMissingCredentials() {
+
+	configPath, err := ioutil.TempFile(os.TempDir(), "s3cmd-")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer os.Remove(configPath.Name())
+
+	_, err = loadConfigFile(configPath.Name())
+	assert.EqualError(suite.T(), err, "failed to find credentials in configuration file")
+}
+
+func (suite *TestSuite) TestConfigMissingEndpoint() {
+	var confFile = `
+access_token = someToken
+access_key = someUser
+`
+	configPath, err := ioutil.TempFile(os.TempDir(), "s3cmd-")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer os.Remove(configPath.Name())
+
+	err = ioutil.WriteFile(configPath.Name(), []byte(confFile), 0644)
+	if err != nil {
+		log.Fatalf("failed to write temp config file, %v", err)
+	}
+
+	_, err = loadConfigFile(configPath.Name())
+	assert.EqualError(suite.T(), err, "failed to find endpoint in configuration file")
+}
```

Comment:
maybe we should also test missing value i.e. 
```
secret_key = 
```

Reply:
That's a good point. Maybe I should have a readme with the expected file instead. What do you think?